numpy/lib/histograms.py

```diff
@@ -217,23 +217,25 @@ def _get_outer_edges(a, range):
     Determine the outer bin edges to use, from either the data or the range
     argument
     """
-    if range is None:
-        if a.size == 0:
-            # handle empty arrays. Can't determine range, so use 0-1.
-            first_edge, last_edge = 0.0, 1.0
-        else:
-            first_edge, last_edge = a.min() + 0.0, a.max() + 0.0
+    if range is not None:
+        first_edge, last_edge = range
+    elif a.size == 0:
+        # handle empty arrays. Can't determine range, so use 0-1.
+        first_edge, last_edge = 0, 1
     else:
-        first_edge, last_edge = [mi + 0.0 for mi in range]
+        first_edge, last_edge = a.min(), a.max()
+
     if first_edge > last_edge:
         raise ValueError(
             'max must be larger than min in range parameter.')
     if not (np.isfinite(first_edge) and np.isfinite(last_edge)):
         raise ValueError(
             'range parameter must be finite.')
+
+    # expand empty range to avoid divide by zero
     if first_edge == last_edge:
-        first_edge -= 0.5
-        last_edge += 0.5
+        first_edge = first_edge - 0.5
+        last_edge = last_edge + 0.5
```

Comment:
Modifying in place is a bad idea, in case `first_edge` and `last_edge` are the same object. This change is needed to pass tests, now that there's no `+ 0.0` to make copies of 0d arrays.